sdk/metric/internal/x/x.go

```diff
@@ -0,0 +1,59 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package x contains support for OTel metric SDK experimental features.
+package x // import "go.opentelemetry.io/otel/sdk/metric/internal/x"
+
+import (
+	"os"
+	"strings"
+)
+
+const EnvKeyRoot = "OTEL_GO_X_"
+
+var (
+	CardinalityLimit = Feature{
+		EnvKeySuffix: "CARDINALITY_LIMIT",
+		// TODO: support accepting number values here to set the cardinality
+		// limit.
+		EnablementVals: []string{"true"},
+	}
+)
+
+type Feature struct {
+	// EnvKeySuffix is the environment variable key suffix the xFeature is
+	// stored at. It is assumed EnvKeyRoot is the base of the environment
+	// variable key.
+	EnvKeySuffix string
+	// EnablementVals are the case-insensitive comparison values that indicate
+	// the Feature is enabled.
+	EnablementVals []string
+}
+
+// Enabled returns if the Feature is enabled.
+func Enabled(f Feature) bool {
```

Comment:
Suggestion:

What if instead of `Enabled` we had functions that transformed the Env to the expected format like `Int()` or `String()`.

I would think the return for these would be `(bool, int)` or `(bool, string)`, where the bool is presence and it's not false/no/0, and the other is a parse of the value if possible.

Reply:
I had intended this package to be a "feature flag" type package. All of the `Feature`s would be boolean types. They would either be enable or disabled or enabled.

The configuration of these experimental features would use the standard OTel environment variables. These environment variables would need to have their values parsed into usable types. I had planned to keep that configuration with the implementation of the feature itself though.

Reply:
Please take a look at https://github.com/open-telemetry/opentelemetry-go/pull/4715. I think it should provide a similar solution as you described.